src/common/utxobased/engine/types.ts

```diff
@@ -3,8 +3,14 @@ import { EdgeSpendInfo } from 'edge-core-js/types'
 
 import { Input, Output } from '../keymanager/utxopicker/types'
 
+// Normalizes a URL to always be a websocket URL
+export const asWebsocketUrl = (raw: unknown): string => {
+  const url = asString(raw)
+  return `${url}/websocket`
```

Comment:
You could use the safer regexp approach to only add this if needed, as isn’t the deleted code.